### ISSUE TITLE
Unify and fix testing of .plot() methods

### ISSUE THREAD

**Issue by adonath** [gammapy/gammapy#1745]
Currently we are testing the `.plot()` methods of our data classes, just by calling `.plot()` and in some cases by calling `mpl_savefig_check()` after. When multiple plot methods are executed in one test session, the plots are drawn on the same axes, because we typically call `plt.gca()` to get the current axes object. This leads to unexpected warnings and `funny` plots such as:

![test](https://user-images.githubusercontent.com/3715928/44595085-5370cf80-a7c8-11e8-94d4-6d76d1845550.png)

I discussed with @cdeil we should introduce a `check_mpl()` context manager that deletes current axes objects in `__enter__` and calls the savefig check in `__exit__`. All places in Gammapy where we test plot methods should the be adapted accordingly.

**Comment by cdeil** [comment]
> context manager that deletes current axes objects in __enter__ and calls the savefig check in __exit__

I think we should do `plt.figure` to make a new figure in `__enter__` instead of deleting existing things, and close it in `__exit__`. See example and discussion in  https://github.com/matplotlib/matplotlib/issues/5218/

> All places in Gammapy where we test plot methods should the be adapted accordingly.

You can get a complete list of all existing plot test with:
```
$ ack requires_dependency | grep matplotlib
```

Of course it would be nice if you could also check somehow if we have completely untested plot methods left; this is a good start:
```
$ ack def | grep plot | egrep -v test
```
Although that also comes out of missing test coverage, so we'll get to that later anyways in a systematic fashion.

Thanks for doing this cleanup!

**Comment by adonath** [comment]
Resolved by #1746.
